dask_sql/physical/rel/custom/create_table_as.py

```diff
@@ -32,29 +34,52 @@ class CreateTableAsPlugin(BaseRelPlugin):
     Nothing is returned.
     """
 
-    class_name = "com.dask.sql.parser.SqlCreateTableAs"
+    class_name = ["CreateMemoryTable", "CreateView"]
 
-    def convert(
-        self, sql: "org.apache.calcite.sql.SqlNode", context: "dask_sql.Context"
-    ) -> DataContainer:
-        schema_name, table_name = context.fqn(sql.getTableName())
+    def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContainer:
+        # Rust create_memory_table instance handle
+        create_memory_table = rel.create_memory_table()
+
+        # can we avoid hardcoding the schema name?
+        schema_name, table_name = context.schema_name, create_memory_table.getName()
 
         if table_name in context.schema[schema_name].tables:
-            if sql.getIfNotExists():
+            if create_memory_table.getIfNotExists():
                 return
-            elif not sql.getReplace():
+            elif not create_memory_table.getOrReplace():
                 raise RuntimeError(
                     f"A table with the name {table_name} is already present."
                 )
 
-        sql_select = sql.getSelect()
-        persist = bool(sql.isPersist())
```

Comment:
I'm interested in how this would've been called in a standard `c.sql(...)` query? The only place throughout the tests where I've seen `PERSIST` used is with Fugue, but I assumed this was being handled on Fugue's end.

Reply:
This information actually comes from the java/calcite side of things. 
https://github.com/dask-contrib/dask-sql/commit/8abc48ce324af7fea0c5fb363a26fb7a70f1a252#diff-258722e0cdeaf803a15b6933e9575641e2d83264e0d8c4cd5cb2ecdd000e74b5R113 
If you see the snippet here, things were set up so that `persist` is true if the user uses the `CREATE TABLE as` syntax but set to false if using the `create_view` syntax.

Reply:
We can follow the same pattern here if preferred or leave it as is and make it configurable in a followup pr based on some discussion here: https://github.com/dask-contrib/dask-sql/issues/269

Reply:
I think for now I will follow the original pattern, but add a TODO to indicate that there is ongoing discussion around the behavior.